report_worker/registry.py

```diff
@@ -0,0 +1,164 @@
+"""
+Report Registry
+
+This module provides a registry system for different report generators.
+Each report type has a unique key and associated generator function.
+"""
+
+import logging
+from typing import Dict, Callable, Any, Optional
+from abc import ABC, abstractmethod
+
+class ReportGeneratorBase(ABC):
+    """Base class for all report generators"""
+    
+    @abstractmethod
+    async def generate(self, job_id: str, organization_id: str, parameters: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Generate a report and return metadata about the generated artifact.
+        
+        Args:
+            job_id: Unique identifier for the report job
+            organization_id: Organization requesting the report
+            parameters: Report-specific parameters
+            
+        Returns:
+            Dict containing:
+            - blob_url: URL to the generated report artifact
+            - file_name: Name of the generated file
+            - file_size: Size of the file in bytes
+            - content_type: MIME type of the generated file
+            - metadata: Any additional metadata about the report
+        """
+        pass
+
+class SampleReportGenerator(ReportGeneratorBase):
+    """Sample report generator for demonstration purposes"""
+    
+    async def generate(self, job_id: str, organization_id: str, parameters: Dict[str, Any]) -> Dict[str, Any]:
+        """Generate a sample report"""
+        import json
+        from datetime import datetime
+        from azure.storage.blob import BlobServiceClient
+        import os
+        
+        # Create sample report content
+        report_content = {
+            "report_type": "sample",
+            "job_id": job_id,
+            "organization_id": organization_id,
+            "generated_at": datetime.utcnow().isoformat(),
+            "parameters": parameters,
+            "data": {
+                "message": "This is a sample report",
+                "status": "completed"
+            }
+        }
+        
+        # Convert to JSON
+        json_content = json.dumps(report_content, indent=2)
+        file_name = f"sample_report_{job_id}.json"
+        
+        # Upload to blob storage
+        connection_string = os.getenv("AZURE_STORAGE_CONNECTION_STRING")
+        if not connection_string:
+            raise ValueError("AZURE_STORAGE_CONNECTION_STRING environment variable not set")
+            
+        blob_service_client = BlobServiceClient.from_connection_string(connection_string)
+        container_name = "reports"
+        blob_name = f"{organization_id}/{file_name}"
+        
+        # Ensure container exists
+        try:
+            container_client = blob_service_client.get_container_client(container_name)
+            container_client.get_container_properties()
+        except Exception:
```

Comment:
Container creation check catches broad Exception

Why it matters: except Exception: hides real problems.

Fix: Catch explicit storage exceptions and handle races cleanly.